### ISSUE TITLE
Incorrect Traced Graph for Softmax

### ISSUE THREAD

**Issue by chughtapan** [pytorch/pytorch#5330]
Hi,
Consider the following example - 
```
class TestSoftMax(nn.Module):
    def __init__(self, input_size, hidden_size):
        super(TestSoftMax, self).__init__()
        self.W = nn.Linear(input_size, hidden_size)

    def forward(self, input):
        return F.softmax(self.W(input))
```
On tracing, the graph generated is 

```
graph(%0 : Float(8, 1024)
      %1 : Float(1024, 1024)
      %2 : Float(1024)
      -------- stage 1 --------
      %7 : Float(8, 1024)) {
  %3 : Float(1024!, 1024!) = t(%1)
  %4 : Float(8!, 1024) = expand[size=[8, 1024]](%2)
  %5 : Float(8, 1024) = addmm[beta={1}, alpha={1}](%4, %0, %3)
  %6 : Float(8, 1024) = softmax[dim=1](%5)
  ---------------- stage 1 ----------------
  %8 : Float(8, 1024) = Constant[value=<Tensor>]()
  %9 : Float(8, 1024) = softmax_backward[dim=1](%7, %5, %8)
  %10 : Float(1024!, 8!) = t(%9)
  %11 : Float(1024, 1024) = mm(%10, %0)
  %13 : Float(1024) = sum[dim=0, keepdim=0](%9)
  return (%6, %11, %13);
}
```
The correct inputs for `%9`  should be `%7,%5,%6`

Thanks

**Comment by apaszke** [comment]
Please use the issue format when submitting them. It includes information that is crucial to fix the problems. We can't do anything until we know e.g. what PyTorch version are you running

**Comment by chughtapan** [comment]
My apologies. 
PyTorch version (https://github.com/pytorch/pytorch/commit/06408168e6d2c071f8d3e14f143118267205f5d0f)

Code to trace the model
```
input_size = 1024
hidden_size = 1024
batch_size = 8
input = torch.autograd.Variable(torch.rand(batch_size, input_size), requires_grad=False).cuda()
L = torch.autograd.Variable(torch.rand(batch_size, hidden_size), requires_grad=False).cuda()
test = TestSoftMax(input_size, hidden_size).cuda()
trace, out = torch.jit.trace(test, input, nderivs=1)
torch.autograd.backward([out], [L])
torch._C._jit_pass_lint(trace)
torch._C._jit_pass_cse(trace)
torch._C._jit_pass_peephole(trace)
torch._C._jit_pass_dce(trace)
graph = trace.graph()
print (graph)
```
